services/QuillLMS/client/app/bundles/Shared/components/lessons/lessonForm.tsx

```diff
@@ -0,0 +1,279 @@
+    questionType: requiredQuestionType || 'questions',
+    selectedQuestions: currentQuestions || [],
+  });
+
+  const handleSubmit = () => {
+    const { name, selectedQuestions, landingPageHtml, flag, modelConceptUID, isELL, questionType } = formState;
+    submit({
+      flag,
+      isELL,
+      landingPageHtml,
+      modelConceptUID,
+      name,
+      questionType,
+      questions: selectedQuestions,
+    });
+  };
+
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFormState(prev => ({ ...prev, name: e.target.value }));
+  }
+
+  const handleQuestionChange = (value: string) => {
+    const { selectedQuestions, questionType } = formState;
+    const changedQuestion = selectedQuestions.find(q => q.key === value);
+    let newSelectedQuestions: any;
+
+    if (!changedQuestion) {
+      newSelectedQuestions = selectedQuestions.concat([{ key: value, questionType }]);
+    } else {
+      newSelectedQuestions = selectedQuestions.filter(question => question !== changedQuestion);
+    }
+    setFormState(prev => ({ ...prev, selectedQuestions: newSelectedQuestions }));
+  };
+
+  const handleSearchChange = ({ value }) => {
+    handleQuestionChange(value);
+  };
+
+  const sortCallback = (sortInfo: any) => {
+    const newOrder = sortInfo.map(item => {
+      return { key: item.key, questionType: item.props.defaultValue };
+    });
+    setFormState(prev => ({ ...prev, selectedQuestions: newOrder }));
+  };
+
+  const renderQuestionSelect = () => {
+    const { selectedQuestions } = formState;
+    if (selectedQuestions && selectedQuestions.length) {
+      const questionsList = selectedQuestions.map((question: { key: string; questionType: string }) => {
+        const dataSource = { questions, fillInBlank, titleCards };
+        const questionObject = dataSource[question.questionType].data[question.key];
+        const prompt = questionObject ? questionObject.prompt : 'Question No Longer Exists';
+        const promptOrTitle = questionObject && question.questionType === 'titleCards' ? questionObject.title : prompt;
+        return (
+          <p className="sortable-list-item" defaultValue={question.questionType} id={question.key} key={question.key}>
+            {promptOrTitle}
+            {'\t\t'}
+            <button onClick={() => handleQuestionChange(question.key)} type="button">Delete</button>
+          </p>
+        );
+      });
+      return <SortableList data={questionsList} id="currently-selected-questions" key={selectedQuestions.length} sortCallback={sortCallback} />;
+    } else {
+      return <div id="currently-selected-questions">No questions</div>;
+    }
+  };
+
+  const renderSearchBox = () => {
+    const { questionType } = formState;
+    const dataSource = { questions, fillInBlank, titleCards };
+    let options = hashToCollection(dataSource[questionType].data);
+    const conceptsData = concepts.data[0];
+    if (options.length > 0) {
+      if (questionType !== 'titleCards') {
```

Comment:
should we just combine this to `if(options.length > 0 && questionType !== 'titleCards')`?